x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/trsuted_apps_grid/index.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useEffect } from 'react';
+import {
+  EuiTablePagination,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiProgress,
+  EuiIcon,
+  EuiBasicTableProps,
+  EuiText,
+} from '@elastic/eui';
+
+import { TrustedApp } from '../../../../../../../common/endpoint/types';
+import { MANAGEMENT_PAGE_SIZE_OPTIONS } from '../../../../../common/constants';
+
+import {
+  getCurrentLocationPageIndex,
+  getCurrentLocationPageSize,
+  getListErrorMessage,
+  getListItems,
+  getListTotalItemsCount,
+  isListLoading,
+} from '../../../store/selectors';
+
+import {
+  useTrustedAppsNavigateCallback,
+  useTrustedAppsSelector,
+  useTrustedAppsStoreActionCallback,
+} from '../../hooks';
+
+import { NO_RESULTS_MESSAGE } from '../../translations';
+
+import { TrustedAppCard } from '../trusted_app_card';
+
+export interface PaginationBarProps {
+  pagination: EuiBasicTableProps<TrustedApp>['pagination'];
+  onPageSizeChange: (pageSize: number) => void;
+  onPageChange: (pageIndex: number) => void;
+}
+
+const PaginationBar = ({ pagination, onPageSizeChange, onPageChange }: PaginationBarProps) => {
+  const pageCount = pagination ? Math.ceil(pagination.totalItemCount / pagination.pageSize) : 0;
+
+  useEffect(() => {
+    if (pagination && pageCount > 0 && pageCount < pagination.pageIndex + 1) {
+      onPageChange(pageCount - 1);
+    }
+  }, [pageCount, onPageChange, pagination]);
+
+  return (
+    <div>
+      <EuiTablePagination
+        activePage={pagination?.pageIndex}
+        hidePerPageOptions={pagination?.hidePerPageOptions}
+        itemsPerPage={pagination?.pageSize}
+        itemsPerPageOptions={pagination?.pageSizeOptions}
+        pageCount={pageCount}
+        onChangeItemsPerPage={onPageSizeChange}
+        onChangePage={onPageChange}
+      />
+    </div>
+  );
+};
+
+export const TrustedAppsGrid = memo(() => {
+  const pageIndex = useTrustedAppsSelector(getCurrentLocationPageIndex);
+  const pageSize = useTrustedAppsSelector(getCurrentLocationPageSize);
+  const totalItemCount = useTrustedAppsSelector(getListTotalItemsCount);
+  const listItems = useTrustedAppsSelector(getListItems);
+  const isLoading = useTrustedAppsSelector(isListLoading);
+  const error = useTrustedAppsSelector(getListErrorMessage);
+
+  const handleTrustedAppDelete = useTrustedAppsStoreActionCallback((trustedApp) => ({
```

Comment:
I resolved all other conversations and let's keep one thread here.

Cause my feeling is that code using pure useCallback to do history navigation and redux action dispatch becomes unnecessarily bulky (prettier is not making it prettier :) )

I would appreciate if we try this out. The whole goal is to write concise code that is not hard to navigate. Currently one single callback that doesn't do much more than navigating url by changing parameters, occupies more than 6 lines.

I also don't like the idea of having to disable the eslint rules, but I've googled for this quite a bit and there are similar stories like this on the git hub page for this rule. In the end we disable eslint rules for even more minor cases which cost almost no effort to do correctly.

If you are worried about memoization of variables from lexical scope then it's true that you can use those in memoization and I could add dependencies list, but I don't think that's the goal of this hook to depend on scope - all that could come from scope is abstracted away (I mean current location) so that you are not supposed to worry about that. It should only get params from event and use those.

In the end this is like ~10 lines of code using this hook? and only in trusted apps (which is pretty small codebase). If it doesn't work out it's an half an hour job (if not less) to replace them.

Reply:
@efreeti my preference leans towards using `useCallback()` since it's an established pattern and a built in hook, so there's some familiarity as well.  Unless there's a strong reason to do otherwise, I don't see a need to break from the pattern just yet.

Also, since we're right on the line and we'd like to ship this in 7.10, I'd suggest we save the conversation for later.  You've built a great view here - let's get it in!

@paul-tavares what are your thoughts?